include/brainclouds2s.h

```diff
@@ -24,8 +35,84 @@ namespace BrainCloud {
     using S2SCallback = std::function<void(const std::string &)>;
     using S2SContextRef = std::shared_ptr<S2SContext>;
 
-    void s2s_log(const std::stringstream &message, bool file = false);
-    void s2s_log(const std::string &message, bool file = false);
+    extern std::string g_logFilePath;
+
+    //void s2s_log(const std::string &message, bool file = false);
+    //void s2s_log(const std::string& message);
+    template<typename ...Args>
+    std::string buildLogMessage(Args && ...args)
+    {
+        std::string timeStamp = TimeUtil::currentTimestamp();
+
+        std::ostringstream oss;
+        oss << "[" << timeStamp << "] ";
+        // Fold expression (not available in C++11), so we expand manually
+        using expander = int[];
+        (void)expander {
+            0, ((void)(oss << std::forward<Args>(args)), 0)...
+        };
+        return oss.str();
+    }
+
+    template <typename... Args>
+    void s2s_log(Args&&... args)
+    {
+        std::string text = buildLogMessage(std::forward<Args>(args)...);
+
+        //Always print to console
+        std::cout << text << std::endl << std::flush;
```

Comment:
Do we always want to print to console? I'm wondering if we should only do so if a logfile hasn't been specified? I know this was the old behaviour but what do we think?